File: CPP/6 Kyu/pyramid_array.cpp
#include <vector>
#include <cstdlib>

std::vector<int> lol(std::size_t n){
  
  std::vector<int> lista = {};
  for(size_t i = 1; i<=n;i++){
      lista.push_back(1);
    }
  return lista;
}

std::vector<std::vector<int>> pyramid(std::size_t n) {
  if(n==0){
    return {};
  }
  

  std::vector<std::vector<int>>lista = {};
    for(size_t i = 1; i<=n;i++){
      lista.push_back(lol(i));
    }
    return lista;
  
}

File: Rust/6 Kyu/pyramid_array.rs

fn lol(n:usize) -> Vec<u32>{
    let mut lista = vec![];
    for x in 1..n+1{
        lista.push(1);
    }
    lista
    
}

fn pyramid(n: usize) -> Vec<Vec<u32>> {
    if n==0{
        return vec![] as Vec<Vec<u32>>;
    }
    let mut lista:Vec<Vec<u32>> = vec![];
    
    for x in 1 ..n+1{
        lista.push(lol(x));
    }
    lista
    
    
    
}